lang/python/wiredtiger.i

```diff
@@ -205,42 +205,40 @@ from packing import pack, unpack
         SWIG_exception_fail(SWIG_AttributeError,
           "bad string value for WT_ITEM");
     $1 = &item;
- }
-
-/* Why do we need an explicit conversion for pl_get_package_part - doesn't the previous typemap cover this? */
-%typemap(in) struct __wt_item *package_buffer (WT_ITEM item) {
-    memset(&item, 0, sizeof(item));
-    if (unpackBytesOrString($input, &item.data, &item.size) != 0)
-        SWIG_exception_fail(SWIG_AttributeError,
-          "bad string value for WT_ITEM");
-    $1 = &item;
- }
-
-/* Special handling for plh_get return arg */
-%typemap(in,numinputs=0) (WT_ITEM *package_result) (WT_ITEM item) {
-    memset(&item, 0, sizeof(item));
-    $1 = &item;
- }
-
-%typemap(argout) (WT_ITEM *package_result) {
-    WT_ITEM *item = $1;
-    PyBytesObject *pbo = PyBytes_FromStringAndSize(item->data, item->size);
-    free(item->data);
-    item->data = NULL;
-    $result = pbo;
 }
 
-/* Special handling for pl_get_package_part return arg */
-%typemap(in,numinputs=0) (WT_ITEM *result) (WT_ITEM item) {
-    memset(&item, 0, sizeof(item));
-    $1 = &item;
+/*
+ * This typemap removes the three last arguments for plh_get, and uses local variables for them instead.
+ * The local variables will be used in the matching argout typemap.
+ * Code in this typemap appears before the call to the API function.
+ */
+%typemap(in,numinputs=0) (WT_ITEM *results_array, u_int *results_count)
+  (WT_ITEM results[32], u_int count) {
+    memset(&results, 0, sizeof(results));
+    count = 32;
+    $1 = results;
+    $2 = &count;
```

Comment:
Can we assert or throw an error if count is larger than 32?

Reply:
I'm adding an assert in the block manager read_multiple for this.